### PR TITLE
`flow-typed` CLI + "validate-defs" command

### PR DESCRIPTION
**Pull request by jeffmo**
This PR does a few things:

1. Changes the `flow-typed` npm package to be a CLI (for now -- `postinstall` script depends on being certain about the structure of the definition repo).
1. Adds infrastructure for a command-CLI to the `flow-typed` npm package that can have commands added as-needed.
1. Adds a `flow-typed validate-defs` command that will go through the definitions repo and verify that everything is structured/named/etc properly. The idea is that we'd run this command -- along with a [yet-to-be-written] `flow-typed run-tests` command -- in Travis for each pull request.
1. Fixes up the existing version dirnames to match the more strict validation requirements (happy to consider relaxing this level of validation, but figured it would be easier to relax later than to be more strict later)

Assuming consensus on this approach (and after code review), I'll begin working on the `flow-typed run-tests` command and wire it into travis.

### CODE REVIEW

**Comment by samwgoldman** [comment]
Would be cool to have non-`asdf` commit messages, assuming this repo is just going to merge on GitHub and not go through the OSS bot which replaces commit messages with PR titles.

**Comment by jeffmo** [comment]
@samwgoldman yea, will update. Still used to internal workflow :)

**Comment by samwgoldman** [comment]
FWIW, GitHub will automatically use the title/description from the commit if a PR is a single commit.

File: npm/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "prepublish": "mkdir -p dist; babel ./src --out-dir=./dist"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/splodingsocks/FlowTyped.git"
+  },
+  "keywords": [
+    "flowtype"
+  ],
+  "author": "Dan Vanderkam (danvdk@gmail.com)",
```
Comment (splodingsocks) [inline npm/package.json]: Do you know who this is?

File: npm/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "bin": "dist/cli.js",
+  "scripts": {
+    "flow": "flow; [[ $?=0||$?=2 ]]",
+    "watch": "mkdir -p dist; babel --watch=./src --out-dir=./dist",
+    "build": "mkdir -p dist; babel ./src --out-dir=./dist",
+    "prepublish": "mkdir -p dist; babel ./src --out-dir=./dist"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/splodingsocks/FlowTyped.git"
```
Comment (splodingsocks) [inline npm/package.json]: Should probably update this url. I can do it in a bit.

Comment (splodingsocks) [inline npm/package.json]: Looks like I can't do it after all. I'm not allowed!

**Comment by splodingsocks** [comment]
@jeffmo Looks fantastic! Thanks for doing this.

What should the merge policy be here? In general reviewers +1 and requesters merge when they're ready?

If so, :+1:

File: npm/package.json (hunk 2 above)
Comment (jeffmo) [inline npm/package.json]: Confirming: You mean you can't commit to the repo?

File: npm/package.json (hunk 1 above)
Comment (jeffmo) [inline npm/package.json]: I don't, wasn't paying close attention... Was it somebody who was contributing before?

File: npm/package.json (hunk 2 above)
Comment (splodingsocks) [inline npm/package.json]: No I think I just don't have push access to your branch. I was going to edit the file directly on Github. I have been able to merge PRs, though, so I think I have write access to the repo.

**Comment by jeffmo** [comment]
> What should the merge policy be here? In general reviewers +1 and requesters merge when they're ready?

Yea, that makes the most sense to me

File: npm/package.json (hunk 1 above)
Comment (samwgoldman) [inline npm/package.json]: cc @danvk

Comment (jeffmo) [inline npm/package.json]: Oh wait I totally know who this is! cc @danvk

Comment (splodingsocks) [inline npm/package.json]: Sounds great.

**Pull request merged**